File: backend/src/main/scala/domains/problemset/objects/request/AddProblemToProblemSetRequest.scala
package domains.problemset.objects.request


import domains.problem.objects.ProblemSlug
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

/** 添加题单题目请求体，携带要加入题单的题目 slug。 */
final case class AddProblemToProblemSetRequest(
  problemSlug: ProblemSlug
)

/** 提供添加题单题目请求体 JSON codec。 */
object AddProblemToProblemSetRequest:
  given Encoder[AddProblemToProblemSetRequest] = deriveEncoder[AddProblemToProblemSetRequest]
  given Decoder[AddProblemToProblemSetRequest] = deriveDecoder[AddProblemToProblemSetRequest]

File: frontend/src/objects/problemset/request/AddProblemToProblemSetRequest.ts
import type { ProblemSlug } from '@/objects/problem/ProblemSlug'

/** 向题集追加题目的请求体；目标题集由 API path 指定。 */
export type AddProblemToProblemSetRequest = {
  problemSlug: ProblemSlug
}
